### PR TITLE
chore(Redesign Servizi): [IARS-20,IARS-21] Introduces networking types and store definition for services preference

### PR DESCRIPTION
**Pull request by CrisTofani**
## Short description
This PR introduces the store definition for services preference and app types.

### CODE REVIEW

File: ts/store/reducers/entities/services/currentService.ts
Comment (CrisTofani) [inline ts/store/reducers/entities/services/currentService.ts:33]:
```suggestion
export const serviceIDCurrentSelector = createSelector(
```

**Review by pietro909 — APPROVED** [review]

File: ts/store/reducers/entities/services/currentService.ts
Comment (Undermaken) [inline ts/store/reducers/entities/services/currentService.ts:17]: imho a new explicit action should be created for this scope, e.g `currentService`

```typescript
export const currentSelectedService = createStandardAction(
  "CURRENT_SELECTED_SERVICE"
)<ServiceID>();
```
it could be dispatched just before where `ROUTES.SERVICE_DETAIL` is dispatched (if FF is enabled)

changing the state using a not explicit action could be dangerous: we know that a service is selected because a specific navigation action is dispatched (this is an implicit/derived info)

furthermore this reducer evaluates the statement `if(action.routeName === ROUTES.SERVICE_DETAIL)` when every navigation action is dispatched

File: ts/types/services/ServicePreference.ts
Comment (Undermaken) [inline ts/types/services/ServicePreference.ts:6]:
```suggestion
export type ServiceID = string & IUnitTag<"ServiceID">;
```
The existing definition is not enough?
`definitions/backend/ServiceId.ts`

File: ts/types/services/ServicePreference.ts
Comment (Undermaken) [inline ts/types/services/ServicePreference.ts:11]: We already have a similar definition
https://github.com/pagopa/io-app/blob/ac6e9ef811711a60d8f6cc5733051bb5c2b07788/ts/utils/profile.ts#L101
Could we use only one of them?

File: ts/store/reducers/entities/services/servicePreference.ts
Comment (Undermaken) [inline ts/store/reducers/entities/services/servicePreference.ts:16]: Do we need to store the preference of multiple services?
Because if we need this info only when a service detail is displayed we could have only the single details instead of a list, e.g `currentServicePreference`

Remember that this section of the store (`entities`) is persisted so there is no deletion logic it will increase without any limits

File: ts/store/reducers/entities/services/servicePreference.ts
Comment (Undermaken) [inline ts/store/reducers/entities/services/servicePreference.ts:35]:
```suggestion
        value: action.payload
```

File: ts/store/reducers/entities/services/servicePreference.ts
Comment (Undermaken) [inline ts/store/reducers/entities/services/servicePreference.ts:61]:
```suggestion
       spById[id] ?? pot.none
```

File: ts/store/reducers/entities/services/servicePreference.ts
Comment (Undermaken) [inline ts/store/reducers/entities/services/servicePreference.ts:20]: Could you leave a comment? 😀

File: ts/types/services/ServicePreferenceResponse.ts
Comment (Undermaken) [inline ts/types/services/ServicePreferenceResponse.ts:4]:
```suggestion
 * Type representing service preference successfully loaded
```

File: ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts
Comment (Undermaken) [inline ts/sagas/services/servicePreference/handleUpsertServicePreferenceSaga.ts:19]:
```suggestion
      value: action.payload
```

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/sagas/services/watchLoadServicesSaga.ts
Comment (CrisTofani) [reply ts/sagas/services/watchLoadServicesSaga.ts:60]: It was a CP & paste error the takeEvery is already present at https://github.com/pagopa/io-app/pull/3177/files#diff-94227b8d19d9a1811d0f31059295a8df3c3007e313489cbc1be29bdaf30b2cdbR37

Comment (CrisTofani) [reply ts/sagas/services/watchLoadServicesSaga.ts:60]: will remove the doubled code

File: ts/types/services/ServicePreference.ts
Comment (CrisTofani) [reply ts/types/services/ServicePreference.ts:11]: didn't know about this! I'll use it here

File: ts/store/reducers/entities/services/servicePreference.ts
Comment (CrisTofani) [reply ts/store/reducers/entities/services/servicePreference.ts:35]: Payload actually contains `id` too, i preferred to avoid redundant information

File: ts/types/services/ServicePreference.ts
Comment (CrisTofani) [reply ts/types/services/ServicePreference.ts]: You're right!

File: ts/store/actions/services/index.ts
Comment (Undermaken) [inline ts/store/actions/services/index.ts:82]:
```suggestion
// User selects a service to view its details
```

**Review by Undermaken — APPROVED** [review]

**Comment by Undermaken** [comment]
@pietro909 waiting for your updated approval

**Review by pietro909 — APPROVED** [review]

**Pull request merged**
